compiler/src/dotty/tools/dotc/transform/VCInlineMethods.scala

```diff
@@ -92,7 +92,7 @@ class VCInlineMethods extends MiniPhase with IdentityDenotTransformer {
       tree // The rewiring will be handled by a fully-applied parent node
     case _ =>
       if (isMethodWithExtension(tree.symbol))
-        rewire(tree).ensureConforms(tree.tpe)
+        rewire(tree).ensureConforms(tree.tpe).withSpan(tree.span)
```

Comment:
I would put the withSpan in the definition of rewire

Reply:
Thanks @smarter . That was what @nicolasstucki and I initially did -- but soon we find it's not the right place due to the recursive nature of `rewire`.

Reply:
There is one branch of `rewire` which is not recursive (the last one), that branch should get the withSpan.

Reply:
The problem is that we want to set the initial `tree.span`, instead of the `tree.span` from recursive calls.

Reply:
The issue there is that the `tree` is the tree of the qualifier which does not have the position of the full call (it does not cover the arguments). We could pass the position as a parameter to `rewire` instead.

Reply:
ah I see, one way to do that would be to change rewire to contain an inner method which is recursive, but I guess it's not very important since it's only used once anyway.

Reply:
Or we could pass the `span` as an extra parameter.